File: backend-php/models/Message.php
<?php
/**
 * Entité Message.
 * Équivalent PHP de com.ihecride.models.Message.
 */

class Message implements JsonSerializable {

    public ?int    $id = null;
    public ?int    $expediteurId = null;
    public ?int    $destinataireId = null;
    public ?string $expediteurNom = null;
    public ?string $contenu = null;
    public ?string $dateEnvoi = null;
    public bool    $lu = false;

    public function jsonSerialize(): array {
        return [
            'id'             => $this->id,
            'expediteurId'   => $this->expediteurId,
            'destinataireId' => $this->destinataireId,
            'expediteurNom'  => $this->expediteurNom,
            'contenu'        => $this->contenu,
            'dateEnvoi'      => $this->dateEnvoi,
            'lu'             => $this->lu,
        ];
    }
}

File: backend/src/main/java/com/ihecride/models/Message.java
package com.ihecride.models;

import java.sql.Timestamp;

public class Message {
    private int id;
    private int expediteurId;
    private int destinataireId;
    private String expediteurNom;
    private String contenu;
    private Timestamp dateEnvoi;
    private boolean lu;

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    public int getExpediteurId() { return expediteurId; }
    public void setExpediteurId(int expediteurId) { this.expediteurId = expediteurId; }
    public int getDestinataireId() { return destinataireId; }
    public void setDestinataireId(int destinataireId) { this.destinataireId = destinataireId; }
    public String getExpediteurNom() { return expediteurNom; }
    public void setExpediteurNom(String expediteurNom) { this.expediteurNom = expediteurNom; }
    public String getContenu() { return contenu; }
    public void setContenu(String contenu) { this.contenu = contenu; }
    public Timestamp getDateEnvoi() { return dateEnvoi; }
    public void setDateEnvoi(Timestamp dateEnvoi) { this.dateEnvoi = dateEnvoi; }
    public boolean isLu() { return lu; }
    public void setLu(boolean lu) { this.lu = lu; }
}
